js/src/framework.ts

```diff
@@ -260,6 +260,29 @@ function initExperiment<IsOpen extends boolean = false>(
   });
 }
 
+export async function callEvaluatorData<
```

Comment:
The idea here is that we have to call `data` (if it's a function) in order to determine whether it's a `BaseExperiment`. We do this by calling the function (and `await`ing it, if it's a Promise), prior to registering the experiment and invoking `runEvaluator`. Is this fine? With the way things are laid out, we can't really get away with _not_ calling the function here if we want to check it, but we _could_ avoid awaiting the promise if we tweak the type definitions.

Or we could just avoid changing the "order of operations" altogether by assigning the base experiment in `runEvaluator` using a separate API call (maybe add a new API endpoint for updating the base experiment?). I was trying to avoid moving stuff around too much but maybe that's cleaner than trying to manipulate `data` before running the evaluator.

Reply:
Expanded the `EvalData` type to allow us to call the `data` fn before initializing the experiment.

Reply:
Does this function still need to be async?